app/Filament/Resources/GameResource/Pages/Edit.php

```diff
@@ -18,6 +20,46 @@ protected function mutateFormDataBeforeSave(array $data): array
     {
         $this->authorizeFields($this->record, $data);
 
+        if (isset($data['ImageIcon'])) {
+            $data['ImageIcon'] = (new ProcessUploadedImageAction())->execute(
+                $data['ImageIcon'],
+                ImageUploadType::GameBadge,
+            );
+        } else {
+            // If no new image was uploaded, retain the existing image.
+            unset($data['ImageIcon']);
+        }
+
+        if (isset($data['ImageTitle'])) {
+            $data['ImageTitle'] = (new ProcessUploadedImageAction())->execute(
+                $data['ImageTitle'],
+                ImageUploadType::GameTitle,
+            );
+        } else {
+            // If no new image was uploaded, retain the existing image.
+            unset($data['ImageTitle']);
+        }
+
+        if (isset($data['ImageIngame'])) {
+            $data['ImageIngame'] = (new ProcessUploadedImageAction())->execute(
+                $data['ImageIngame'],
+                ImageUploadType::GameInGame,
+            );
+        } else {
+            // If no new image was uploaded, retain the existing image.
+            unset($data['ImageIngame']);
+        }
+
+        if (isset($data['ImageBoxArt'])) {
+            $data['ImageBoxArt'] = (new ProcessUploadedImageAction())->execute(
+                $data['ImageBoxArt'],
+                ImageUploadType::GameBoxArt,
+            );
+        } else {
+            // If no new image was uploaded, retain the existing image.
+            unset($data['ImageBoxArt']);
+        }
```

Comment:
May be able to cut down a lot of duplication with:

```php
$imageTypes = [
    'ImageIcon' => ImageUploadType::GameBadge,
    'ImageTitle' => ImageUploadType::GameTitle, 
    'ImageIngame' => ImageUploadType::GameInGame,
    'ImageBoxArt' => ImageUploadType::GameBoxArt,
];

foreach ($imageTypes as $field => $type) {
    if (isset($data[$field])) {
        $data[$field] = (new ProcessUploadedImageAction())->execute($data[$field], $type);
    } else {
        // If no new image was uploaded, retain the existing image.
        unset($data[$field]);
    }
}
```

Reply:
I like it. 

I wish there were some way to turn the inner lines of code into a function, as the `if isset {} else unset` occurs every place the `ProcessUploadedImageAction` is used.